### PR TITLE
v3: workbox-range-requests module

### PR DESCRIPTION
**Pull request by jeffposnick**
R: @addyosmani @gauntface

Fairly straightforward port of the `workbox-range-requests` code from v2, with some test changes to accommodate the mocked interfaces, and some additional error messages.

I updated the `workbox-sw` mapping, but I might be missing some other "meta" things that need to be changed when we add a new library.

### CODE REVIEW

File: packages/workbox-range-requests/Plugin.mjs [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+import {createPartialResponse} from './createPartialResponse.mjs';
+
+import './_version.mjs';
+
+/**
+ * A class implementing the `cachedResponseWillBeUsed` lifecycle callback.
```
Comment (gauntface) [inline packages/workbox-range-requests/Plugin.mjs]: This description should cover the purpose of the class, not the implementation details.

> The range request plugin makes it easy for a Request with a 'range' header to be forfilled by a cached response.
> It does this by intercepting the `cachedResponseWillBeUsed` plugin callback and returning the appropriate subset of the cached Response body.

File: packages/workbox-range-requests/Plugin.mjs [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+ *
+ * @memberof workbox.rangeRequests
+ */
+class Plugin {
+  /**
+   * @param {Object} options
+   * @param {Request} options.request The original request, which may or may not
+   * contain a Range: header.
+   * @param {Response} options.cachedResponse The complete cached response.
+   * @return {Promise<Response>} If request contains a Range: header, then
```
Comment (gauntface) [inline packages/workbox-range-requests/Plugin.mjs]: Nit, could you shorten this to: "If request contains a 'Range' header, then a new Response with status 206 whose body is a subset of `cachedResponse` is returned, otherwise `cachedResponse` is returned as-is."

Comment (jeffposnick) [reply packages/workbox-range-requests/Plugin.mjs]: Done.

File: packages/workbox-range-requests/Plugin.mjs (hunk 1 above)
Comment (jeffposnick) [reply packages/workbox-range-requests/Plugin.mjs]: Done.
